### PR TITLE
Fixes #705: Fix tests on Ruby 3.0

### LINKED ISSUE #705: Ruby 3.0 support
Test suite fails running against Ruby 3.0. There is very likely needed PR #698, but even then there are errors such as:

~~~
  84) UsersController Parameter processing / extraction ignore not described parameters
      Failure/Error: ret = super(*args)
      ArgumentError:
        wrong number of arguments (given 2, expected 1)
      # /usr/share/gems/gems/actionpack-6.0.3.4/lib/action_controller/test_case.rb:460:in `process'
      # ./lib/apipie/extractor/recorder.rb:154:in `process'
      # ./spec/spec_helper.rb:16:in `block in process'
      # ./spec/spec_helper.rb:29:in `compatible_request'
      # ./spec/spec_helper.rb:16:in `process'
      # /usr/share/gems/gems/rails-controller-testing-1.0.5/lib/rails/controller/testing/template_assertions.rb:62:in `process'
      # /usr/share/gems/gems/actionpack-6.0.3.4/lib/action_controller/test_case.rb:403:in `post'
      # /usr/share/gems/gems/rails-controller-testing-1.0.5/lib/rails/controller/testing/integration.rb:16:in `block (2 levels) in <module:Integration>'
      # ./spec/controllers/users_controller_spec.rb:743:in `block (3 levels) in <top (required)>'
~~~

It seems the crux is in these two methods:
https://github.com/Apipie/apipie-rails/blob/cdff4ec654ee70dd222c7d75f64e20b167b44fce/spec/spec_helper.rb#L15-L33
introduced in 8fb9ecb8d078ea99bcdb054b5997955ccd33e08b.

First of all, the `method` and `action` parameters are swapped, but since they are swapped consistently, it does not cause any harm.

But secondly, I don't think I can really untangle the compatibility matrix between all rails and ruby version. There should be possibly used everywhere parameters such as `*args, **kwargs`, but I'm not sure how to convert the current code :confused:

### PR DESCRIPTION
**Pull request by hank-spokeo**

### CODE REVIEW

**Comment by jsantos** [comment]
Any chances of getting this merged? Thanks!

**Comment by Sylor-huang** [comment]
Any chances of getting this merged? Thanks! @hank-spokeo  @ofedoren

**Comment by jsantos** [comment]
Bump?

**Comment by suarez-agu** [comment]
Any news on this? It would be great to have Ruby 3.0 support. @hank-spokeo @ofedoren 

Thanks!

**Comment by mathieujobin** [comment]
@hank-spokeo Thanks for this PR 

can you rebase and move up your changes from the travis file to the github/action script ? 

thanks

**Comment by hank-spokeo** [comment]
@mathieujobin @ofedoren PR updated, added Ruby 3.1 as well

**Review by mathieujobin — APPROVED** [review]
Wonderful !

File: lib/apipie/extractor/recorder.rb [hunk 1]
```diff
@@ -160,6 +160,7 @@ def process(*, **) # action, parameters = nil, session = nil, flash = nil, http_
         ensure
           Apipie::Extractor.clean_call_recorder
         end
+        ruby2_keywords :process if respond_to?(:ruby2_keywords, true)
```
Comment (mathieujobin) [inline lib/apipie/extractor/recorder.rb:163]: Can you explain to us why you added this? And why it causes a warning?

Thanks

Comment (hank-spokeo) [reply lib/apipie/extractor/recorder.rb:163]: See https://github.com/Apipie/apipie-rails/pull/716/commits/684cd533d27261b36b7cfea06584191f39eabafa

I think what happened is https://github.com/Apipie/apipie-rails/commit/a55d836f4e8609395c374d85b2542a3ac7277a2c got merged, but the `process` params should be

```
        def process(*) # action, parameters = nil, session = nil, flash = nil, http_method = 'GET')
          ret = super
```
per https://eregon.me/blog/2021/02/13/correct-delegation-in-ruby-2-27-3.html

Comment (jsantos) [reply lib/apipie/extractor/recorder.rb:163]: Thanks @hank-spokeo! I'll create a PR to fix this.

Comment (mathieujobin) [reply lib/apipie/extractor/recorder.rb:163]: Thanks a lot @hank-spokeo

**Pull request merged**